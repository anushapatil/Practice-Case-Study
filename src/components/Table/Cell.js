import React from 'react'

const Cell = ({ data }) => {
  return (
    <div class="Cell">
      <p>{data}</p>
    </div>
  )
}

export default Cell
