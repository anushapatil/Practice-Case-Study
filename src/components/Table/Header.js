import React from 'react'

const headers = [ 'ROUTE', 'DESTINATION', 'DEPARTS']

const Header = () => {
  return (
    <div class="Heading">
      {headers.map((data, index) =>
        <div class="HeaderCell" key={index}>
          <p>{data}</p>
        </div>
      )}
    </div>
  )
}

export default Header
