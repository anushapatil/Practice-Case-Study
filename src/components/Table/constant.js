export const NO_DEPARTURE = 'No departures at this time'
export const ROUTE = 'ROUTE'
export const DESTINATION = 'DESTINATION'
export const DEPARTS = 'DEPARTS'
