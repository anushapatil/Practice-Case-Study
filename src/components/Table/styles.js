import styled from 'styled-components'

const TableStyle = styled.div`
  .Table {
    display: table;
    color: #626462;
    font-size: 1.125rem !important;
  }
  .Title {
    display: table-caption;
    text-align: center;
    font-weight: bold;
    font-size: larger;
  }
  .Heading {
    display: table-row;
    font-weight: bold;
    padding-top: 12px;
    padding-bottom: 12px;
    text-align: left;
  }
  .HeaderCell {
    background-color: #ffd200;
    display: table-cell;
    padding: 8px;
  }
  .Row {
    display: table-row;
  }
  .Cell {
    display: table-cell;
    padding: 0.75rem;
    padding-left: 1.5rem !important;
    border-top: none;
    border-bottom: none;
    letter-spacing: 1.28px;
  }
  .Cell:nth-child(even) {
    background-color: #f5f5f4;
  }
  .RouteCell {

  }
  .DestinationCell {

  }
  .DepartsCell {

  }
`
export default TableStyle
