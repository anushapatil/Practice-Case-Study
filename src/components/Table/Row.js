import React, { Fragment } from 'react'

import { NO_DEPARTURE } from './constant'

const Row = ({ list }) => {
  if (list.length === 0) return <div className='first-col'>{NO_DEPARTURE}</div>
  return (
    <Fragment>
      {list && list.map(({ route_short_name = '', terminal = '', description = '', departure_text = '' }, index) =>
        <div key={index} className='row border-bottom ml-0 mr-0'>
          <div className='col-3 first-col p-3 text-left font-weight-bold'>
            {route_short_name + terminal}
          </div>
          <div className='col-5 second-col p-3 text-left'>
            {description}
          </div>
          <div className='col-4 last-col p-3 text-right font-weight-bold'>
            {departure_text}
          </div>
        </div>
      )}
    </Fragment>
  )
}

export default Row
