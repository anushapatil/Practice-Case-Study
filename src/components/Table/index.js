import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import Header from './Header'
import Row from './Row'
import TableStyle from './styles'

const Table = ({
  departures,
  departsDescription,
  departsStopId
}) => {
  return (
    <TableStyle className='container'>
      <div className='title d-flex'>
        <h3>{departsDescription}</h3>
        <span>
          <strong>
            Stop #: {departsStopId}
          </strong>
        </span>
      </div>
      <Header />
      <Row list={departures} />
    </TableStyle>
  )
}

Table.propTypes = {
  departures: PropTypes.array,
  departsDescription: PropTypes.string,
  departsStopId: PropTypes.number
}

const mapStateToProps = (state) => ({
  departures: state.transit.departures,
  departsDescription: state.transit.departsDescription,
  departsStopId: state.transit.departsStopId
})

export default connect(mapStateToProps)(Table)