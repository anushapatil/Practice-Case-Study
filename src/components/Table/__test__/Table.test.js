import React from 'react'
import configureStore from 'redux-mock-store'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'

import Table from '../'

const mockStore = configureStore([])

// UI testing
test('Snapshot Table', () => {
  const store = mockStore({
    transit: {
      departures: [{
        actual: true,
        departure_text: '6 Min',
        departure_time: 1607571120,
        description: 'to Mpls-Target Field',
        direction_id: 1,
        direction_text: 'WB',
        route_id: '902',
        route_short_name: 'Green',
        schedule_relationship: 'Scheduled',
        stop_id: 56026,
        trip_id: '17807892-DEC20-RAIL-Weekday-01'
      }], 
      departsDescription: 'MOA station',
      departsStopId: 1
    }
  })

  store.dispatch = jest.fn()

  const appComponent = renderer.create(
    <Provider store={store}>
      <Table />
    </Provider>
  )
  expect(appComponent.toJSON()).toMatchSnapshot()
})