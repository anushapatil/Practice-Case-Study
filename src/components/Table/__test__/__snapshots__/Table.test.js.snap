// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Table 1`] = `
<div
  className="sc-AykKC MrlJI pl-0 pr-0 pb-3 bg-light bg-secondary"
>
  <div
    className="title d-flex p-3"
  >
    <h4>
      MOA station
    </h4>
    <span>
      <strong>
        Stop #:
      </strong>
      1
    </span>
  </div>
  <div
    className="row ml-0 mr-0 bg-warning"
  >
    <div
      className="col-3 p-3 text-left font-weight-bold"
    >
      ROUTE
    </div>
    <div
      className="col-5 p-3 text-left font-weight-bold"
    >
      DESTINATION
    </div>
    <div
      className="col-4 p-3 text-right font-weight-bold"
    >
      DEPARTS
    </div>
  </div>
  <div
    className="row border-bottom ml-0 mr-0"
  >
    <div
      className="col-3 first-col p-3 text-left font-weight-bold"
    >
      Green
    </div>
    <div
      className="col-5 second-col p-3 text-left"
    >
      to Mpls-Target Field
    </div>
    <div
      className="col-4 last-col p-3 text-right font-weight-bold"
    >
      6 Min
    </div>
  </div>
</div>
`;
